core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1490,4 +1490,29 @@ object TestUtils extends Logging {
       Metrics.defaultRegistry.removeMetric(metricName)
   }
 
+  def stringifyTopicPartitions(partitions: Set[TopicPartition]): String = {
+    Json.legacyEncodeAsString(
+      Map(
+        "version" -> 1,
+        "partitions" -> partitions.map(tp => Map("topic" -> tp.topic, "partition" -> tp.partition))
+      )
+    )
+  }
+
+  def resource[R <: AutoCloseable, A](resource: R)(func: R => A): A = {
+    var error: Throwable = null
+
+    try {
+      func(resource)
+    } catch {
```

Comment:
This catch-finally seems weird. coming from java, dropping catch and removing throw from finally would give the same result. Isn't its same case in Scala?

Reply:
Yes. It has the same semantic as Java. I was originally planning to catch the exception from `resource.close()` and add it as suppressed to the exception from `func(resource)` but decided against it. I changed the code to a `try { ... } finally { ... }` as you suggested.